src/images/avatar.ts

```diff
@@ -8,6 +9,36 @@ import {Component, Input, ChangeDetectionStrategy} from 'angular2/core';
 export class NglAvatar {
   @Input() src: string = '';
   @Input() alt: string = '';
-  @Input() size: 'x-small' | 'small' | 'medium' | 'large' = 'medium';
-  @Input() type: 'rectangle' | 'circle' = 'rectangle';
+  @Input() size: 'x-small' | 'small' | 'medium' | 'large';
+
+  @Input('size') set setSize(value: 'x-small' | 'small' | 'medium' | 'large') {
+    this.updateClass(this._size, value);
+    this._size = value;
+  }
+
+  @Input('type') set setType(value: 'rectangle' | 'circle') {
+    this.updateClass(this._type, value);
+    this._type = value;
+  }
+
+  private _type: 'rectangle' | 'circle';
+  private _size: 'x-small' | 'small' | 'medium' | 'large';
+
+  constructor(public element: ElementRef, public renderer: Renderer) {
+    renderer.setElementClass(element.nativeElement, 'slds-avatar', true);
+  }
+
+  ngOnInit() {
+    if (!this._type) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--rectangle', true);
+    }
+
+    if (!this._size) {
+      this.renderer.setElementClass(this.element.nativeElement, 'slds-avatar--medium', true);
+    }
+  }
+
+  private updateClass(oldValue: string, newValue: string) {
+    replaceClass(this, `slds-avatar--${oldValue}`, newValue ? `slds-avatar--${newValue}` : '');
+  }
 };
```

Comment:
@bekos, yes, it fails but I believe this is a result of how `NgClass` works. A little digging showed that the first time, the classes are rendered properly. But if the expression given to `NgClass` is not something computed or something that changes so as to get re-computed, `NgClass` does not re-apply the class (i.e., `NgClass` thinks it's the only *class-binding* player). `[class]="var_name"` seems to work though even if `var_name` is always the same value.

This means that if you were to add another class specification bound to anything computed, `custom-class` and the other class specification would both be working fine.

Actually, in this case the most proper thing for a user would be to simply say `class="foo"` instead of `[ngClass]="'foo'"`, which works. The reasoning here is that we are using a styling directive, using a constant expression, which I think is too verbose.

So the trade-off here is verbosity vs making sure that the user will not use `NgClass` where the expression is constant. I may not like how `NgClass` works (in contrast to `[class]`), but since the user will *definitely* choose the *not-good-for-us* path (given the choice :smile:), I would go with verbosity.

P.s., this is a **bummer** because I really liked the short version! :disappointed: